components/sftp/sftp.app.js

```diff
@@ -0,0 +1,70 @@
+module.exports = {
+  type: "app",
+  app: "sftp",
+  props: {
+    db: "$.service.db"
+  },
+  methods: {
+    config: {
+      host: null,
+      port: null,
+      username: null,
+      password: null,
+    },
+    async getClient(deps) {
+      const ssh2SftpClient = require('ssh2-sftp-client'); // src: https://github.com/theophilusx/ssh2-sftp-client/blob/master/src/index.js
```

Comment:
Generally I'd recommend moving these requires up to the top of the file for clarity / convention.